spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ArrowVectorAccessors.java

```diff
@@ -0,0 +1,505 @@
+      return new DecimalAccessor((IcebergArrowVectors.DecimalArrowVector) vector);
+    } else if (vector instanceof IcebergArrowVectors.VarcharArrowVector) {
+      return new StringAccessor((IcebergArrowVectors.VarcharArrowVector) vector);
+    } else if (vector instanceof IcebergArrowVectors.VarBinaryArrowVector) {
+      return new BinaryAccessor((IcebergArrowVectors.VarBinaryArrowVector) vector);
+    } else if (vector instanceof DateDayVector) {
+      return new DateAccessor((DateDayVector) vector);
+    } else if (vector instanceof TimeStampMicroTZVector) {
+      return new TimestampAccessor((TimeStampMicroTZVector) vector);
+    } else if (vector instanceof ListVector) {
+      ListVector listVector = (ListVector) vector;
+      return new ArrayAccessor(listVector);
+    } else if (vector instanceof StructVector) {
+      StructVector structVector = (StructVector) vector;
+      return new StructAccessor(structVector);
+    }
+    throw new UnsupportedOperationException("Unsupported vector: " + vector.getClass());
+  }
+
+  private static class BooleanAccessor extends ArrowVectorAccessor {
+
+    private final BitVector vector;
+
+    BooleanAccessor(BitVector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final boolean getBoolean(int rowId) {
+      return vector.get(rowId) == 1;
+    }
+  }
+
+  private static class IntAccessor extends ArrowVectorAccessor {
+
+    private final IntVector vector;
+
+    IntAccessor(IntVector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final int getInt(int rowId) {
+      return vector.get(rowId);
+    }
+  }
+
+  private static class LongAccessor extends ArrowVectorAccessor {
+
+    private final BigIntVector vector;
+
+    LongAccessor(BigIntVector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final long getLong(int rowId) {
+      return vector.get(rowId);
+    }
+  }
+
+  private static class DictionaryLongAccessor extends ArrowVectorAccessor {
+
+    private final Dictionary parquetDictionary;
+    private final IntVector offsetVector;
+
+    DictionaryLongAccessor(IntVector vector, Dictionary dictionary) {
+      super(vector);
+      this.offsetVector = vector;
+      this.parquetDictionary = dictionary;
+    }
+
+    @Override
+    final long getLong(int rowId) {
+      return parquetDictionary.decodeToLong(offsetVector.get(rowId));
+    }
+  }
+
+  private static class FloatAccessor extends ArrowVectorAccessor {
+
+    private final Float4Vector vector;
+
+    FloatAccessor(Float4Vector vector) {
+      super(vector);
+      this.vector = vector;
+    }
+
+    @Override
+    final float getFloat(int rowId) {
+      return vector.get(rowId);
+    }
+  }
+
+  private static class DictionaryFloatAccessor extends ArrowVectorAccessor {
+
+    private final IntVector offsetVector;
+    private final Dictionary parquetDictionary;
```

Comment:
This updated `DictionaryDoubleAccessor` to decode into an array, but the other types weren't similarly updated. I think they should be for consistency and so it is clear what's happening.